ecommerce/core/utils.py

```diff
@@ -72,3 +79,632 @@ def use_read_replica_if_available(queryset):
+
+    Returns:
+        str: A formatted predicate string of the form:
+            'attributes.`brand-text` in ("org1", "org2", "org3")'
+
+    Examples:
+        >>> extract_orgs('org:(MITx OR HarvardX)')
+        'attributes.`brand-text` in ("HarvardX","MITx")'
+
+    Notes:
+        - Supports organization values with or without parentheses or quotes.
+        - Case-insensitive handling of logical operators (AND, OR).
+        - Removes duplicate organization values from the output.
+    """
+    # Regular expression to find values after 'org:'
+    org_pattern = re.findall(r'org:\s*(?:\((.*?)\)|"(.*?)"|([\w]+))', text)
+
+    # Extract and clean orgs from the captured groups
+    orgs = [org.strip().replace('"', '') for group in org_pattern for org in group if org]
+
+    # Split on AND/OR (case-insensitive) and flatten the list
+    split_orgs = [item for org in orgs for item in re.split(r'\s+(?:AND|OR)\s+', org, flags=re.IGNORECASE)]
+
+    # Removing duplicates and joining as a single string
+    orgs_str = ",".join(sorted(set(f'"{org}"' for org in split_orgs)))
+
+    return f'attributes.`brand-text` in ({orgs_str})'
+
+
+def _process_org_values(input_string):
+    """
+    Processes organization values from the input string and generates a SQL-like predicate.
+
+    This function identifies and processes organization values following the 'org:' keyword.
+    It handles both positive and negative organization values, supporting multiple formats:
+    - Parentheses with AND/OR operators (e.g., org:(A OR B AND -C))
+    - Quoted strings (e.g., org:"A")
+    - Plain values (e.g., org:A)
+
+    Negative organization values (preceded by '-') are processed using the `not in` clause, while
+    positive values are processed using the `in` clause.
+
+    Args:
+        input_string (str): Input query string containing organization values.
+
+    Returns:
+        str: A formatted SQL-like predicate string:
+            - For negative values: 'attributes.`brand-text` not in ("org1", "org2", ...)'
+            - For positive values: 'attributes.`brand-text` in ("org1", "org2", ...)'
+            - If no matching orgs are found: "No org values found"
+
+    Examples:
+        >>> process_org_values('org:(-MITx OR HarvardX)')
+        'attributes.`brand-text` not in ("MITx")'
+
+        >>> process_org_values('org:(HarvardX AND MITx)')
+        'attributes.`brand-text` in ("HarvardX","MITx")'
+
+    Notes:
+        - Supports both positive and negative org values.
+        - Handles logical operators (AND, OR) case-insensitively.
+        - Removes duplicate values in the output.
+        - If no org values are found, returns "No org values found".
+    """
+    # Regular expression to match org values
+    pattern = r'org:\s*(?:\((.*?)\)|"(.*?)"|([\w]+))'
+
+    # Find all matching org groups
+    matches = re.findall(pattern, input_string)
+
+    if not matches:
+        return "No org values found"
```

Comment:
Add info log for no org found if needed
```suggestion
        return None
```

Reply:
I have made change related to this